source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+
+var onednngraph = onednngraph || {};
+var json = json || require('./json');
+var base = base || require('./base');
+
+onednngraph.ModelFactory = class {
+
+    match(context) {
+        const identifier = context.identifier;
+        const extension = identifier.split('.').pop().toLowerCase();
+        if (extension === 'json') {
+            const obj = context.open('json');
+            if (obj && obj.version && obj.graph) {
```

Comment:
Multiple JSON formats can define `version` and `graph` properties, add unique approach to identify the format?

Reply:
added engine_kind and fpmath_mode as identifier.

Reply:
`this._runtime = symbol.engine_kind + ' ' + symbol.fpmath_mode`?

Reply:
`this._runtime = obj.engine_kind + ' ' + obj.fpmath_mode;` should be used in `onednn.Model` constructor instead of `this._runtime = 'engine: ' + symbol.engine_kind + '; fpmath: ' + symbol.fpmath_mode + ';';`.

Reply:
Is the version check `if (!Object.prototype.hasOwnProperty.call(symbol, 'version')) {` actually required? The code seems to work without it.

Reply:
Same with `if (!symbol) {` which never evaluates to `true`.